src/monodroid/jni/android-system.cc

```diff
@@ -772,7 +769,7 @@ AndroidSystem::copy_file_to_internal_location (char *to_dir, char *from_dir, cha
 }
 #endif /* defined (ANDROID) */
 
-int
+long
```

Comment:
This was done presumably because `max_gref_count` was changed to `long`, but -- as per above https://github.com/xamarin/xamarin-android/pull/3500/files#r315469031 -- I don't understand *why* this should be `long`.

Additionally, the return expression casts to `int`; if we *are* changing to `long`, shouldn't the return expression use `static_cast<long>(...)`?

Reply:
`int` is defined to be at least **16** bits long, while `long` is defined to be at least **32** bits long. One of the compilers (I think it was mingw) complained about it and `long` is the safest, common denominator, value here.